src/factory/factory.cpp

```diff
@@ -81,6 +83,30 @@ Factory::CreateIndex(const std::string& origin_name,
                                              params.use_opq,
                                              params.use_bsa,
                                              params.use_async_io);
+        } else if (name == INDEX_HGRAPH) {
+            auto json_param = nlohmann::json::parse(parameters);
+            IndexCommonParam param;
+            CHECK_ARGUMENT(json_param.contains(PARAMETER_METRIC_TYPE),
```

Comment:
use CHECK_ARGUMENT in parsing process rather than here.

Reply:
done